src/main/io/serial_cli.c

```diff
@@ -1710,6 +1710,13 @@ void cliProcess(void)
             clicmd_t target;
             cliPrint("\r\n");
 
+            // Strip comment starting with # from line
+            char *p = cliBuffer;
+            p = strchr(++p, '#');
```

Comment:
You can start from `p` here and discard whole line if # is first character. Then it is possible to remove '#' test later ...